### ISSUE TITLE
Crash in Bitmap index scan with ORCA

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#6274]
This is a variant of a test in 'aocs' test file. I started seeing this crash, when I introduced a bug in the ANALYZE code, to compute pg_class.reltuples incorrectly. That changed the plan that ORCA generates for
this query, into one that trips an assertion:

```
create table bms_ao_bug
(c1 int not null,
c2 int not null,
c3 char(100) not null)
with (appendonly=true, orientation=column, compresstype=zlib)
distributed by (c1)
;

insert into bms_ao_bug select 1, 1, a.c1::char(100) from generate_series(1, 2000000) a(c1);

create index bms_ao_bug_ix1 on bms_ao_bug (c2);

set enable_seqscan=off;
set work_mem=256;

-- Change the stats, to force different plan:
set allow_system_table_mods=true;
update pg_class set reltuples=100 where relname = 'bms_ao_bug';

select
a.c1,
count(*) row_cnt
from bms_ao_bug a
where a.c2 = 1
group by
a.c1
;

ERROR:  Unexpected internal error (execBitmapAOScan.c:267)  (seg2 slice1 127.0.0.1:40002 pid=22295) (execBitmapAOScan.c:267)
DETAIL:  FailedAssertion("!(((bool) (((const void*)(slot_get_ctid(slot)) != ((void *)0)) && ((slot_get_ctid(slot))->ip_posid != 0))))", File: "execBitmapAOScan.c", Line: 267)
HINT:  Process 22295 will wait for gp_debug_linger=120 seconds before termination.
```

the plan:
```
postgres=# explain select                                                 
a.c1,
count(*) row_cnt
from bms_ao_bug a
where a.c2 = 1
group by
a.c1
;
                                                          QUERY PLAN                                                           
-------------------------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice3; segments: 3)  (cost=0.00..204.47 rows=1 width=12)
   ->  GroupAggregate  (cost=0.00..204.47 rows=1 width=12)
         Group Key: c1
         ->  Sort  (cost=0.00..204.47 rows=1 width=12)
               Sort Key: c1
               ->  Redistribute Motion 3:3  (slice2; segments: 3)  (cost=0.00..204.47 rows=1 width=12)
                     Hash Key: c1
                     ->  Result  (cost=0.00..204.47 rows=1 width=12)
                           ->  GroupAggregate  (cost=0.00..204.47 rows=1 width=12)
                                 Group Key: c1
                                 ->  Sort  (cost=0.00..204.47 rows=34 width=4)
                                       Sort Key: c1
                                       ->  Redistribute Motion 3:3  (slice1; segments: 3)  (cost=0.00..204.47 rows=34 width=4)
                                             ->  Bitmap Table Scan on bms_ao_bug  (cost=0.00..204.47 rows=34 width=4)
                                                   Recheck Cond: (c2 = 1)
                                                   ->  Bitmap Index Scan on bms_ao_bug_ix1  (cost=0.00..0.00 rows=0 width=0)
                                                         Index Cond: (c2 = 1)
 Planning time: 14.203 ms
 Optimizer: PQO version 3.9.0
(19 rows)
```
Without updating the stats, ORCA chooses a Table Scan instead of Bitmap Index Scan.

**Comment by hlinnaka** [comment]
This was fixed on master by PR #6290, but I think it's still an issue on 5X_STABLE. (I'm getting a crash with ORCA on 5X_STABLE on my laptop with this query, but I'm not 100% if it's because of this bug, or something else.)

**Comment by ivannovick** [comment]
not planning the fix on older version
